### ISSUE TITLE
Implement remote locking for xtokens

### ISSUE THREAD

**Issue by Chralt98** [open-web3-stack/open-runtime-module-library#935]
Is there a plan to implement XCMv3 remote locks in orml_xtokens for a simple abstraction?

https://substrate.stackexchange.com/questions/9214/how-does-xcm-remote-locking-work

Would be great to have this, but I am too lazy to do it myself at the moment, because it's not that urgent.

**Comment by xlc** [comment]
No plan yet. Also remote locking won't be very useful for non system parachains as the tokens are locked in our parachains rather assets hub and I dont' think parachain teams will redo all the work to rebuilt the parachains in such way to have the relay tokens stores on assets hub.

**Comment by Chralt98** [comment]
Thanks for the answer. I don't understand. The asset hub is Statemine or Statemint, right? 

What I have in mind is the following:

A user on Karura locks KAR and sets the unlocker to my custom parachain. Now my parachain can query the locked balance and sees that a user locked KAR and do a token holder vote with those KAR on my parachain, without the need of the user to transfer his KAR to my parachain (trustless).

**Comment by xlc** [comment]
Right. I see. That will be a valid use case. But Karura will need to support this remote voting as well, which shouldn't be hard IF the governance pallets provided by Substrate supports remove voting (which I don't see any issues about implementing those).
